pkg/config/template.go

```diff
@@ -73,17 +73,29 @@ type configuration struct {
 	}
 
 	Database struct {
-		ConnectionString      SecureString  `mapstructure:"connection_string"`
-		MaxOpenConnections    int32         `mapstructure:"max_open_connections"`
-		MaxIdleConnections    int32         `mapstructure:"max_idle_connections"`
+		// Deprecated: use Postgres struct
+		ConnectionString SecureString `mapstructure:"connection_string"`
+		// Deprecated: use Postgres struct
+		MaxOpenConnections int32 `mapstructure:"max_open_connections"`
+		// Deprecated: use Postgres struct
+		MaxIdleConnections int32 `mapstructure:"max_idle_connections"`
+		// Deprecated: use Postgres struct
 		ConnectionMaxLifetime time.Duration `mapstructure:"connection_max_lifetime"`
+
 		// KVEnabled Development flag to switch between postgres DB and KV store implementations
 		KVEnabled bool `mapstructure:"kv_enabled"`
 		// DropTables Development flag to delete tables after successful migration to KV
 		DropTables bool `mapstructure:"drop_tables"`
 		// Type  Name of the KV Store driver DB implementation which is available according to the kv package Drivers function
 		Type string `mapstructure:"type"`
 
+		Postgres *struct {
+			ConnectionString      SecureString  `mapstructure:"connection_string"`
```

Comment:
We also need to modify the documentation for these new values.

Reply:
https://github.com/treeverse/lakeFS/pull/3881